File: allin-front-angular/src/app/subsystems/admin/models/commands/create-Person-command.ts
import { ExtendedFieldValueModel } from "../../../shared/models/extended-field-value-model";

export interface CreatePersonCommand {
  firstName?: string; // nvarchar(150)
  lastName?: string;  // nvarchar(150)
  ssn?: string;       // nvarchar(50)
  mobiles?: Mobile[]; // nvarchar(500)
  email?: string;      // nvarchar(50)
  photoUrl?: string;   // nvarchar(500)
  signatureImageUrl?: string; // nvarchar(500)
  birthDate?: Date;    // date
  genderEnum?: number; // bigint
  maritalEnum?: number; // smallint
  isLegal?: boolean;
  personAddresses?: Address[];


  extendedFieldValues?: ExtendedFieldValueModel[];
}

export interface Mobile {
  type: string;
  phoneNumber: string;
}

export interface Address {
  type: number;
  addressLine: string;
}

File: Allin.Backend.Net/Microservices/Admin/Allin.Admin.Application/Commands/Persons/Create/CreatePersonCommand.cs
using Allin.Admin.Infrastructure.Persistence;
using Allin.Common.Utilities.Mappings;
using MediatR;

namespace Allin.Admin.Application.Commands
{
    public class CreatePersonCommand : IRequest<bool>, IMapFrom<Person, CreatePersonCommand>
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Ssn { get; set; }
        public IList<PhoneArg> Mobiles { get; set; }
        public bool? IsLegal { get; set; }
        public string Email { get; set; }
        public string PhotoUrl { get; set; }
        public string SignatureImageUrl { get; set; }
        public DateOnly? BirthDate { get; set; }
        public GenderEnums GenderEnum { get; set; }
        public MaritalEnums? MaritalEnum { get; set; }
        public IEnumerable<ExtendedFieldArg> ExtendedFieldValues { get; set; }
        public IEnumerable<AddressArg> PersonAddresses { get; set; }

    }

    public class ExtendedFieldArg : IMapFrom<TableExtendedFieldValue, ExtendedFieldArg>
    {
        public long TableExtendedFieldId { get; set; }
        public string Value { get; set; }
        public string? ValueFieldItemId { get; set; }
    }

    public class AddressArg : IMapFrom<PersonAddress, AddressArg>
    {
        public long TypeBaseId { get; set; }
        public long CityBaseId { get; set; }
        public string Address { get; set; }
    }

    public class PhoneArg
    {
        public PhoneTypeEnum Type { get; set; }
        public string PhoneNumber { get; set; }

        public enum PhoneTypeEnum
        {
            HomePhone = 1,

            OfficePhone = 2,
            Mobile = 3,
            Fax = 4
        }
    }
}